### ISSUE TITLE
pylint reporting error for pydantic module not exists 

### ISSUE THREAD

**Issue by yinuu** [microsoft/vscode-pylint#366]
### pylint tips: No name 'BaseModel' in module 'pydantic', indeed it is in the right way and I can go for the reference and token color indicate editor can check its location too;  so what's going wrog? Help needed, Thanks for any instruction !

![image](https://github.com/microsoft/vscode-pylint/assets/19586687/39bc2706-8973-4f02-b5a4-6110e29589af)

#### pydantic version and pylint plugin version as belows

```
pydantic           1.10.7      Data validation and settings management usin...
```

![image](https://github.com/microsoft/vscode-pylint/assets/19586687/93b815b7-cbfc-4402-bf93-24be69c9766e)

**Comment by yinuu** [comment]
follow your instructions, here is some information about the case:

>  name         : pylint
 version      : 2.17.4
 description  : python code static checker

dependencies
 - astroid >=2.15.4,<=2.17.0-dev0
 - colorama >=0.4.5
 - dill >=0.3.6
 - isort >=4.2.5,<6
 - mccabe >=0.6,<0.8
 - platformdirs >=2.2.0
 - tomlkit >=0.10.1

#### base.py
```Python
from pydantic import BaseModel as PydanticBaseModel, Extra


class BaseModel(PydanticBaseModel):
    """BaseModel"""

    class Config:
        """Config Overwrite Pydantic default"""

        extra = Extra.forbid


class TestStub(BaseModel):
    """TestStub"""

    name: str
    age: int | None = None

```
pylint app/src/model/base.py

************* Module model.base
app/src/model/base.py:1:0: C0114: Missing module docstring (missing-module-docstring)
app/src/model/base.py:1:0: E0611: No name 'BaseModel' in module 'pydantic' (no-name-in-module)
app/src/model/base.py:7:4: R0903: Too few public methods (0/2) (too-few-public-methods)
app/src/model/base.py:4:0: R0903: Too few public methods (0/2) (too-few-public-methods)
app/src/model/base.py:13:0: R0903: Too few public methods (0/2) (too-few-public-methods)

------------------------------------------------------------------
Your code has been rated at 0.00/10 (previous run: 0.00/10, +0.00)
```

It seems an issue with the pylint library.
